src/adapters/world-name-permission-checker.ts

```diff
@@ -97,3 +119,34 @@ export const createOnChainDclNameChecker = async (
     checkPermission
   }
 }
+
+export const createEndpointNameChecker = async (
+  components: Pick<AppComponents, 'config' | 'logs' | 'fetch'>
+): Promise<IWorldNamePermissionChecker> => {
+  const nameCheckUrl = await components.config.requireString('ENDPOINT_NAME_CHECKER_BASE_URL')
+
+  return {
+    checkPermission: async (ethAddress: EthAddress, worldName: string): Promise<boolean> => {
+      if (worldName.length === 0 || ethAddress.length === 0) {
+        return false
+      }
+      return await components.fetch
+        .fetch(nameCheckUrl, {
+          method: 'POST',
+          body: JSON.stringify({
+            worldName: worldName,
+            ethAddress: ethAddress
+          })
+        })
+        .then((response) => response.json())
```

Comment:
please expand into awaits instead of mixing .then, in order to check for non 200 status codes